exotic/exotic.py

```diff
@@ -1428,24 +1459,119 @@ def stellar_variability(fit_lc_refs, fit_lc_best, comp_stars, vsp_comp_stars, vs
         return []
 
     try:
-        vsp_params = [{
-            'time': lc_fit.jd_times[mask_ref][i],
-            'airmass': lc_fit.airmass[mask_ref][i],
-            'mag': mt,
-            'mag_err': Mt_err[i],
-            'cname': vsp_auid_comp,
-            'cmag': Mc,
-            'pos': comp_pos
-        } for i, mt in enumerate(Mt)]
+        vsp_params = gather_stellar_data(Mt, Mt_err, Mc, lc_fit.airmass[mask_ref], lc_fit.jd_times[mask_ref], target_id,
+                                         vsp_id_comp, comp_pos)
 
-        plot_stellar_variability(vsp_params, save, s_name, vsp_auid_comp)
+        plot_stellar_variability(vsp_params, save, star_name=star_name)
     except Exception as e:
         log_info(f"Error in plotting or finalizing stellar variability data: {e}", warn=True)
         return []
 
     return vsp_params
 
 
+def get_comparison_star_parameters(vsp_params, ref_flux, vsp_list, vsp_comp_stars):
+    position = vsp_params[0]['pos']
+    comp_star = next(
+        vsp_comp_stars[ckey] for ckey in vsp_comp_stars.keys() if position == vsp_comp_stars[ckey]['pos'])
+    vsp_comp_star = ref_flux[vsp_list.index(position)]
+
+    return {
+        'position': position,
+        'times': vsp_comp_star['myfit'].jd_times,
+        'aperture': vsp_comp_star['photometry_parameters']['aperture'] if vsp_comp_star['photometry_method'] == 'Aperture' else None,
+        'annulus': vsp_comp_star['photometry_parameters']['annulus'] if vsp_comp_star['photometry_method'] == 'Aperture' else None,
+        'flux': vsp_comp_star['myfit'].comparison_flux,
+        'airmass': vsp_comp_star['myfit'].airmass,
+        'id': next(key for key, value in vsp_comp_stars.items() if value['pos'] == position),
+        'Mc': comp_star['mag'],
+        'Mc_err': comp_star['error']
+    }
+
+
+def calculate_fortuitous_star_flux(target_key, comparison_star_params, aper_data, psf_data):
+    if comparison_star_params['aperture']:
+        aperture = comparison_star_params['aperture']
+        annulus = comparison_star_params['annulus']
+        target_flux = aper_data[target_key][:, aperture, annulus]
+    else:
+        target_flux = 2 * np.pi * psf_data[target_key][:, 2] * psf_data[target_key][:, 3] * psf_data[target_key][:, 4]
+
+    return target_flux
+
+
+def calculate_airmass_model(norm_flux, comparison_star_params):
+    transformed_norm_flux = np.log(norm_flux)
+
+    X = sm.add_constant(comparison_star_params['airmass'])
+
+    model = sm.OLS(transformed_norm_flux, X).fit()
+
+    a1 = model.params[1]
+    a2 = model.params[0]
+
+    airmass_model = np.exp(a1) * np.exp(a2 * comparison_star_params['airmass'])
+
+    return airmass_model, a2
```

Comment:
I'd like to highlight this function on calculating the airmass model with its coefficients.